### PR TITLE
[Buttons] Support colored shadows

### PR DESCRIPTION
**Pull request by romoore**
Adding new methods to set the shadow color for each UIControlState.

### CODE REVIEW

File: components/Buttons/tests/unit/ButtonsTests.m [hunk 1]
```diff
@@ -45,10 +45,10 @@ static CGFloat randomNumberNotEqualTo(const CGFloat targetNumber) {
 static UIColor *randomColor() {
   switch (arc4random_uniform(5)) {
     case 0:
-      return [UIColor whiteColor];
+      return [UIColor colorWithRed:1 green:1 blue:1 alpha:1];
       break;
     case 1:
-      return [UIColor blackColor];
+      return [UIColor colorWithRed:0 green:0 blue:0 alpha:1];
```
Comment (samnm) [inline components/Buttons/tests/unit/ButtonsTests.m:51]: What is the purpose of these two changes?

File: components/Buttons/src/MDCButton.m [hunk 2]
```diff
@@ -262,6 +270,11 @@ - (void)commonMDCButtonInit {
   shadowLayer.shadowColor = [UIColor blackColor].CGColor;
   shadowLayer.elevation = [self elevationForState:self.state];
 
+  if (!_shadowColors) {
```
Comment (samnm) [inline components/Buttons/src/MDCButton.m]: You don't need this if statement.

**Review by samnm — CHANGES_REQUESTED** [review]

**Comment by ianegordon** [comment]
Were you going to update / add a sample to demonstrate functionality?

File: components/Buttons/tests/unit/ButtonsTests.m (hunk 1 above)
Comment (romoore) [reply components/Buttons/tests/unit/ButtonsTests.m:51]: The new unit test that verifies the layer's actual shadowColor uses `-colorWithCGColor:`, which places the color in the ~~SRGB~~ DeviceRGB color space by default. `[UIColor blackColor]` and `-whiteColor` are in the Grey color space, and so object equality checks failed whenever those colors were used.

File: components/Buttons/src/MDCButton.m (hunk 2 above)
Comment (romoore) [reply components/Buttons/src/MDCButton.m]: If `_shadowColors` was created in `-initWithCoder:`, we shouldn't overwrite it.

File: components/Buttons/tests/unit/ButtonsTests.m [hunk 3]
```diff
@@ -217,6 +217,95 @@ - (void)testBackgroundColorForState {
...
+    if ((effectiveControlState & UIControlStateHighlighted) == UIControlStateHighlighted) {
+      button.highlighted = YES;
+    } else {
+      button.highlighted = NO;
+    }
+
+    if ((effectiveControlState & UIControlStateDisabled) == UIControlStateDisabled) {
+      button.enabled = NO;
+      // Disabling a button turns off "highlighted"
+      effectiveControlState = (controlState & ~UIControlStateHighlighted);
```
Comment (romoore) [inline components/Buttons/tests/unit/ButtonsTests.m]: this should be `effectiveControlState`

File: components/Buttons/src/MDCButton.m (hunk 2 above)
Comment (romoore) [reply components/Buttons/src/MDCButton.m]: Nevermind, I just realized that commonInit is called before the decoding takes place.

File: components/Buttons/src/MDCButton.m [hunk 4]
```diff
@@ -439,6 +453,30 @@ - (void)updateTitleCase {
   }
 }
 
+- (void)updateShadowColor {
+  self.layer.shadowColor = [self shadowColorForState:self.state].CGColor;
+}
+
+- (void)setShadowColor:(UIColor *)shadowColor forState:(UIControlState)state {
+  if(shadowColor) {
```
Comment (chuga) [inline components/Buttons/src/MDCButton.m]: nit: you're missing a space here

Comment (romoore) [reply components/Buttons/src/MDCButton.m]: Done

**Review by samnm — APPROVED** [review]

**Review by ajsecord — APPROVED** [review]

**Pull request merged**